src/DotNetty.Common/Utilities/StringUtil.cs

```diff
@@ -0,0 +1,331 @@
+                    {
+                        if (res[i] == "")
+                        {
+                            res.Remove(res[i]);
+                        }
+                        else
+                        {
+                            break;
+                        }
+                    }
+                }
+            }
+
+            return res.ToArray();
+        }
+
+        /// <summary>
+        /// Get the item after one char delim if the delim is found (else null).
+        /// This operation is a simplified and optimized
+        /// version of {@link String#split(String, int)}.
+        /// </summary>
+        /// <param name="value"></param>
+        /// <param name="delim"></param>
+        /// <returns></returns>
+        public static string SubstringAfter(this string value, char delim)
+        {
+            int pos = value.IndexOf(delim);
+            return pos >= 0 ? value.Substring(pos + 1) : null;
+        }
+
+        /// <summary>
+        ///  Converts the specified byte value into a 2-digit hexadecimal integer.
+        /// </summary>
+        public static string ByteToHexStringPadded(int value)
+        {
+            return Byte2HexPad[value & 0xff];
+        }
+
+        //    /**
+        // * Converts the specified byte value into a 2-digit hexadecimal integer and appends it to the specified buffer.
+        // */
+        //public static <T extends Appendable> T byteToHexStringPadded(T buf, int value) {
+        //    try {
+        //        buf.append(byteToHexStringPadded(value));
+        //    } catch (IOException e) {
+        //        PlatformDependent.throwException(e);
+        //    }
+        //    return buf;
+        //}
+
+        /// <summary>
+        /// Converts the specified byte array into a hexadecimal value.
+        /// </summary>
+        public static string ToHexStringPadded(byte[] src)
+        {
+            return ToHexStringPadded(src, 0, src.Length);
+        }
+
+        /// <summary>
+        /// Converts the specified byte array into a hexadecimal value.
+        /// </summary>
+        public static string ToHexStringPadded(byte[] src, int offset, int length)
+        {
+            int end = offset + length;
+            var sb = new StringBuilder(length << 1);
+            for (int i = offset; i < end; i++)
+            {
+                sb.Append(ByteToHexStringPadded(src[i]));
+            }
+            return sb.ToString();
+        }
+
+        public static StringBuilder ToHexStringPadded(StringBuilder sb, byte[] src, int offset, int length)
+        {
+            int end = offset + length;
+            for (int i = offset; i < end; i++)
+            {
+                sb.Append(ByteToHexStringPadded(src[i]));
+            }
+            return sb;
+        }
+
+        /// <summary>
+        /// Converts the specified byte value into a hexadecimal integer.
+        /// </summary>
+        public static string ByteToHexString(int value)
```

Comment:
The method name is about "byte" but the param is of int type. it seems to be byte instead.